### PR TITLE
Add CONTRIBUTING.md

### LINKED ISSUE #646: Improve the developer documentation
Right now the developer documentation is very sparse:

https://github.com/bids-standard/bids-validator#development

In particular, it assumes that:

1. The developer knows Javascript, and already has it installed
2. The developer understands `npm` and already has it installed
3. The developer understands the general structure of the repository

I think that we cannot assume that any of these is true for the typical "user" audience of the bids-validator. The developer docs could be improved to make this project a bit more welcoming to new contributors who may not be javascript devs.

### LINKED ISSUE #788: Tutorial walking through contributing to BIDS validator
This point was raised during the BIDS community meeting - it could be useful to have a tutorial on how to contribute/extend the BIDS validator (a walkthrough) and tips/tricks that helped contribute. 

@sappelhoff: perhaps may you please write down your process and things that were useful that you learned through the process. This will be really useful to have as BEPs are getting closer to opening a PR and work in parallel with the validator to support the extension. I can help you with this too!

### PR DESCRIPTION
**Pull request by sappelhoff**
closes #788 
closes #646 

This PR improves the developer documentation and provides contributors with a landing page with lots of useful information.

Please advise what you think could be added, improved, or deleted.

Keep in mind: "Don't let the perfect be the enemy of the good" :wink: 

cc @choldgraf @effigies @franklin-feingold @rwblair

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  - git clone
+  - git branch
+  - git checkout
+  - git status
+  - git pull
+  - git add
+  - git commit
+  - git push
+  - [How to configure git for your user](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration)
+1. **Next, familiarize yourself with [GitHub](https://github.com) and how it
```
Comment (rwblair) [inline CONTRIBUTING.md]: Did we want these subsections to be numbered 1,2,3... or to just be bullet points?

**Comment by rwblair** [comment]
A bit about running `yarn test` locally might be nice to see if your changes break any existing functionality. Not sure if it should go in testing section, or earlier in the making changes section.

Good job @sappelhoff this is much needed.

**Comment by effigies** [comment]
I've got a number of suggested fixes that will be easier to do as a PR to your PR.

**Comment by sappelhoff** [comment]
sounds great @effigies just ping me and I'll review and merge it

**Comment by effigies** [comment]
See sappelhoff#1

**Review by effigies — APPROVED** [review]
(Obviously) LGTM. @rwblair Want to have another look?

**Comment by effigies** [comment]
Oh, just saw the discussion of `yarn test`. That sounds reasonable to me.

**Comment by choldgraf** [comment]
Looks really nice - thanks for this one, getting started developing guides are super useful for the community!

**Comment by choldgraf** [comment]
Looks really nice - thanks for this one, getting started developing guides are super useful for the community!

**Comment by rwblair** [comment]
Looks good to me as well.

**Comment by franklin-feingold** [comment]
Thank you for putting this together @sappelhoff ! This will be a great guide and resource for the community!

**Pull request merged**